src/main/scala/mesosphere/marathon/raml/AppConversion.scala

```diff
@@ -308,6 +303,20 @@ trait AppConversion extends DefaultConversions with ConstraintConversion with En
       ))
       else None
 
+    val unreachableStrategy = service.when(_.hasUnreachableStrategy, _.getUnreachableStrategy.toRaml).orElse(App.DefaultUnreachableStrategy)
+
+    val hasPersistentVolumes = service.hasContainer && service.getContainer.getVolumesList.exists(_.hasPersistent)
+
+    val residency = if (hasPersistentVolumes || service.hasResidency) {
+      unreachableStrategy.flatMap {
```

Comment:
Why not match as you've done above? Is `unreachableStrategy` an `Option`? That seems weird.

Also, why not factor this out since it's used in several places?

Reply:
The problem here is that different classes are used. Above we use Raml, and here we use classes from the state package.